datasets/hate_speech_pl/hate_speech_pl.py

```diff
@@ -0,0 +1,112 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""HateSpeech Corpus for Polish"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+
+import datasets
+
+
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = r"""\
+@article{troszynski2017czy,
+  title={Czy komputer rozpozna hejtera? Wykorzystanie uczenia maszynowego (ML) w jako{\'s}ciowej analizie danych},
+  author={Troszy{\'n}ski, Marek and Wawer, Aleksandra},
+  journal={Przegl{\k{a}}d Socjologii Jako{\'s}ciowej},
+  volume={13},
+  number={2},
+  pages={62--80},
+  year={2017},
+  publisher={Uniwersytet {\L}{\'o}dzki, Wydzia{\l} Ekonomiczno-Socjologiczny, Katedra Socjologii~…}
+}
+"""
+
+_DESCRIPTION = """\
+HateSpeech corpus in the current version contains over 2000 posts crawled from public Polish web. They represent various types and degrees of offensive language, expressed toward minorities (eg. ethnical, racial). The data were annotated manually.
+"""
+
+_HOMEPAGE = "http://zil.ipipan.waw.pl/HateSpeech"
+
+_LICENSE = "CC BY-NC-SA"
+
+_URLs = [
+    "https://raw.githubusercontent.com/aiembassy/hatespeech-corpus-pl/master/data/fragment_anotatora_2011_ZK.csv",
+    "https://raw.githubusercontent.com/aiembassy/hatespeech-corpus-pl/master/data/fragment_anotatora_2011b.csv",
+    "https://raw.githubusercontent.com/aiembassy/hatespeech-corpus-pl/master/data/fragment_anotatora_2012_luty.csv",
+]
+
+
+class HateSpeechPl(datasets.GeneratorBasedBuilder):
+    """HateSpeech Corpus for Polish"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("uint16"),
+                    "text_id": datasets.Value("uint32"),
+                    "annotator_id": datasets.Value("uint8"),
+                    "minority_id": datasets.Value("uint8"),
+                    "negative_emotions": datasets.Value("bool"),
+                    "call_to_action": datasets.Value("bool"),
+                    "source_of_knowledge": datasets.Value("uint8"),
+                    "irony_sarcasm": datasets.Value("bool"),
+                    "topic": datasets.Value("uint8"),
```

Comment:
Can you use the ClassLabel feature type for topic please ?
for example
```
"topic": datasets.ClassLabel(names=[topic_name_for_identifier_0, topic_name_for_identifier_1, etc.])
```

you'll have to regenerate the dataset_infos.json file after this change

Reply:
Sure, I was wondering if the ClassLabel should be used here. Actually it could have been applied to many different columns as well.